lib/processors/jsdoc/lib/ui5/plugin.js

```diff
@@ -274,15 +274,17 @@ function analyzeModuleDefinition(node) {
 							.variables.filter((curVar) => prop.value.name === curVar.name)[0];
 						const renamedVar = checkVarRenaming(resolvedVar);
 
-						return renamedVar ? renamedVar.original : prop.value.name;
+						return { original: renamedVar.original, renamed: prop.value.name };
 					}) // ObjectPattern means destructuring of the parameter
-					: [currentModule.factory.params[i].name]; // simple Identifier
+					: [{original: currentModule.factory.params[i].name}]; // simple Identifier
 
 			names.forEach(name => {
 				const module = resolveModuleName(currentModule.module, currentModule.dependencies[i]);
-				debug(`  import ${name} from '${module}'`);
-				currentModule.localNames[name] = {
-					module: module
+				debug(`  import ${name.renamed || name.original} from '${module}'`);
+
+				currentModule.localNames[name.renamed || name.original] = {
+					module: module,
+					property: name.original,
```

Comment:
This has been enhanced but I don't see any usage of the new property. How is the rename/descructuring use case actually solved now?

Reply:
To be honest, I don't exactly remember clearly the reasoning for resolving the variable to its original name. Still, I remember when we decided to keep the name as it had been defined in the code but record the original name as a property for traceability.

I just checked the file history and I believe this change should be the key here: https://github.com/SAP/ui5-builder/commit/cb2e115b2da034f030a2d3fe4774e6a7e2c3cac1

In the beginning, [this change](https://github.com/SAP/ui5-builder/commit/cb2e115b2da034f030a2d3fe4774e6a7e2c3cac1) was resolving properly just local variables, but not destructured (and renamed) function arguments and as far as I remember, that's why I needed to keep the original name of the restructured argument. From the current perspective, this now seems like it was a bug :D 
With the refactoring of the `checkVarRenaming()` and more specifically with [this reducer](https://github.com/SAP/ui5-builder/pull/775/files#diff-693e706995c87056125c1cff2fe8953f9592038c8f50206171a84a6d2244d8ddR701), it is now able to correctly resolve destructuring and renaming of arguments